cpp/src/arrow/util/bit_util.h

```diff
@@ -366,5 +367,18 @@ void PackBits(const uint32_t* values, uint8_t* out) {
   }
 }
 
+/// Determine if a word has a zero byte
+/// The subexpression (v - 0x01010101UL), evaluates to a high bit set in any byte whenever
+/// the corresponding byte in v is zero or greater than 0x80. The sub-expression ~v &
+/// 0x80808080UL evaluates to high bits set in bytes where the byte of v doesn't have its
+/// high bit set (so the byte was less than 0x80). Finally, by ANDing these two
+/// sub-expressions the result is the high bits set where the bytes in v were zero, since
+/// the high bits set due to a value greater than 0x80 in the first sub-expression are
+/// masked off by the second.
+uint64_t HasZeroByte(uint64_t value) {
```

Comment:
Since it's a "has" return `bool`?

Reply:
oh I get the point, I don't know whether putting this in `bit_util` a good idea, since it's like `swisstable`'s logic

Reply:
yes, it will a little bit confusing, move this function to swiss table class and rename it to GetZeroByteMask will be better